scripts/extensions/datetimeField/src/getTemplateDateTimeField.tsx

```diff
@@ -0,0 +1,230 @@
+import {ISuperdesk, ITemplateEditorComponentProps} from 'superdesk-api';
+import * as React from 'react';
+import set from 'date-fns/set';
+import format from 'date-fns/format';
+import addMinutes from 'date-fns/addMinutes';
+import {IDateTimeFieldConfig} from './extension';
+import {DatePickerISO, TimePicker, Button, Switch} from 'superdesk-ui-framework/react';
+import {Radio, CheckGroup} from 'superdesk-ui-framework';
+
+interface ITemplateDateTimeFieldState {
+    radioValue: string | null;
+    previousValue: any;
+}
+
+function isDateValue(value: any) {
+    return !!Date.parse(value);
+}
+
+export function getTemplateDateTimeField(superdesk: ISuperdesk) {
+    const {gettext, gettextPlural} = superdesk.localization;
+    const {getLocaleForDatePicker} = superdesk.ui.framework;
+    const {Spacer} = superdesk.components;
+    const {dateToServerString} = superdesk.utilities;
+    const currentDateTime = '{{ now|iso_datetime }}';
+
+    return class TemplateDateTimeField extends
+        React.PureComponent<ITemplateEditorComponentProps<string | null, IDateTimeFieldConfig>,
+        ITemplateDateTimeFieldState> {
+        constructor(props: any) {
+            super(props);
+
+            this.state = {
+                radioValue: isDateValue(this.props.value) ? this.props.value : 'setCurrentDate',
+                previousValue: new Date(),
+            };
+
+            this.onRadioValueChange = this.onRadioValueChange.bind(this);
+        }
+
+        onRadioValueChange(radioValue: any) {
+            this.setState({radioValue});
+            if (radioValue === 'setCurrentDate') {
+                this.props.setValue(currentDateTime);
+            } else {
+                this.props.setValue(this.state.previousValue);
+            }
+        }
+
+        render() {
```

Comment:
In case user has selected to set a static date-time, can you reuse the component from `src/getDateTimeField.tsx` instead of copy-pasting the code from it?

Reply:
we have not copied exactly the same, if you see [here](https://github.com/superdesk/superdesk-client-core/pull/3864/files/32395021e0da0fe595d2248a60ada727ec232d18#diff-12d88161b515852811fe7dbc60312ec50e13d30c6aeb4a5d4d620c837d15261bR58) we have set some values and have some [checks](https://github.com/superdesk/superdesk-client-core/pull/3864/files/32395021e0da0fe595d2248a60ada727ec232d18#diff-12d88161b515852811fe7dbc60312ec50e13d30c6aeb4a5d4d620c837d15261bR113) as well.

Reply:
I'm not saying you copied exactly, but it's always better to reuse when possible, even if you have to add a few configurable things in the original component(which you might not even need to do here).

Regarding the first example, you don't need to modify the component and can call `onRadioValueChange` in a `setValue` callback:

```
<DateTimeField
    setValue={(val) => {
        if (val == null) {
            // ...
        } else {
            // ...
            this.onRadioValueChange('setCurrentDate');
        }
    }}
/>
```